intermediate_source/speech_command_recognition_with_torchaudio.py

```diff
@@ -0,0 +1,523 @@
+"""
+Speech Command Recognition with torchaudio
+==========================================
+
+This tutorial will show you how to correctly format an audio dataset and
+then train/test an audio classifier network on the dataset.
+
+Colab has GPU option available. In the menu tabs, select “Runtime” then
+“Change runtime type”. In the pop-up that follows, you can choose GPU.
+After the change, your runtime should automatically restart (which means
+information from executed cells disappear).
+
+First, let’s import the common torch packages such as
+``torchaudio <https://github.com/pytorch/audio>``\ \_ that can be
+installed by following the instructions on the website.
+
+"""
+
+# Uncomment the following line to run in Google Colab
+
+# GPU:
+# !pip install torch==1.7.0+cu101 torchvision==0.8.1+cu101 torchaudio==0.7.0 -f https://download.pytorch.org/whl/torch_stable.html
+
+# CPU:
+# !pip install torch==1.7.0+cpu torchvision==0.8.1+cpu torchaudio==0.7.0 -f https://download.pytorch.org/whl/torch_stable.html
+
+# For interactive demo at the end:
+# !pip install pydub
+
+import os
+from base64 import b64decode
+from io import BytesIO
+
+import IPython.display as ipd
+import matplotlib.pyplot as plt
+from tqdm.notebook import tqdm
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+import torchaudio
+from google.colab import output as colab_output
+from pydub import AudioSegment
+from torchaudio.datasets import SPEECHCOMMANDS
+
+######################################################################
+# Let’s check if a CUDA GPU is available and select our device. Running
+# the network on a GPU will greatly decrease the training/testing runtime.
+#
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print(device)
+
+
+######################################################################
+# Importing the Dataset
+# ---------------------
+#
+# We use torchaudio to download and represent the dataset. Here we use
+# SpeechCommands, which is a datasets of 35 commands spoken by different
+# people. The dataset ``SPEECHCOMMANDS`` is a ``torch.utils.data.Dataset``
+# version of the dataset.
+#
+# The actual loading and formatting steps happen in the access function
+# ``__getitem__``. In ``__getitem__``, we use ``torchaudio.load()`` to
+# convert the audio files to tensors. ``torchaudio.load()`` returns a
```

Comment:
Is it worth mentioning the internal mechanism about `torchaudio.load`, which can be changed anytime in the future?

Reply:
My impression was that some details would help the user understand. I agree though this might be too detailed, especially since those are internal details. I've replaced this by the following to instead emphasize that there exists a torchaudio.load function in case someone just wants to load a file instead of using a dataset.

> The actual loading and formatting steps happen when a data point is being accessed, and torchaudio takes care of converting the audio files to tensors. If one wants to load an audio file directly instead, ``torchaudio.load()`` can be used. It returns a tuple containing the newly created tensor along with the sampling frequency of the audio file (16kHz for SpeechCommands).